traffic_ops/testing/api/v4/parameters_test.go

```
@@ -246,3 +261,331 @@ func DeleteTestParameter(t *testing.T, pl tc.Parameter) {
+	if err != nil {
+		t.Errorf("Getting Parameters by Invalid Name %v", err)
+	}
+	if len(resp.Response) >= 1 {
+		t.Errorf("Invalid name shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetParametersByInvalidConfigfile(t *testing.T) {
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("configFile", "invalidfile")
+	resp, _, err := TOSession.GetParameters(opts)
+	if err != nil {
+		t.Errorf("Getting Parameters by Invalid ConfigFile %v", err)
+	}
+	if len(resp.Response) >= 1 {
+		t.Errorf("Invalid config file shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetParametersByInvalidValue(t *testing.T) {
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("value", "invalidvalue")
+	resp, _, err := TOSession.GetParameters(opts)
+	if err != nil {
+		t.Errorf("Getting Parameters by Invalid value %v", err)
+	}
+	if len(resp.Response) >= 1 {
+		t.Errorf("Invalid value shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func CreateTestParametersAlreadyExist(t *testing.T) {
+	opts := client.NewRequestOptions()
+	resp, _, _ := TOSession.GetParameters(opts)
+	if len(resp.Response) > 0 {
+		_, reqInf, _ := TOSession.CreateParameter(resp.Response[0], opts)
+		if reqInf.StatusCode != http.StatusBadRequest {
+			t.Errorf("Expected 400 status code, got %v", reqInf.StatusCode)
+		}
+	} else {
+		t.Errorf("No existing parameters available to validate duplicate functionality")
+	}
+}
+
+func CreateTestParametersMissingName(t *testing.T) {
+	opts := client.NewRequestOptions()
+	if len(testData.Parameters) > 0 {
+		firstParameter := testData.Parameters[0]
+		firstParameter.Name = ""
+		_, reqInf, _ := TOSession.CreateParameter(firstParameter, opts)
+		if reqInf.StatusCode != http.StatusBadRequest {
+			t.Errorf("Expected 400 status code, got %v", reqInf.StatusCode)
+		}
+	}
+}
+
+func CreateTestParametersMissingconfigFile(t *testing.T) {
+	opts := client.NewRequestOptions()
+	if len(testData.Parameters) > 0 {
```

This test will silently pass if the test data is in a bad state and the test cannot proceed

---

Added failure checks